vyper/semantics/namespace.py

```diff
@@ -176,6 +176,8 @@ def validate_identifier(attr):
     # `address` members
     "balance",
     "codesize",
+    "codehash",
```

Comment:
I don't see where `.codehash` was introduced

Reply:
It could be a separate issue, but I noticed that currently (e.g. with vyper 0.3.1) the code like below causes an internal error:

```py
# examples/repros/pr2583.vy 

struct S:
    codehash: Bytes[4]

@external
def foo(s: S) -> Bytes[4]:
  return s.codehash
```

```txt
$ vyper examples/repros/pr2583.vy 
Error compiling: examples/repros/pr2583.vy
vyper.exceptions.TypeCheckFailure: Attribute node did not produce LLL. vyper.ast.nodes.Attribute:
     5 def foo(s: S) -> Bytes[4]:
---> 6   return s.codehash
----------------^
     7

This is an unhandled internal compiler error. Please create an issue on Github to notify the developers.
https://github.com/vyperlang/vyper/issues/new?template=bug.md
```

I wasn't sure if this is supposed to work in the first place, but I suppose the current error is due to the `elif` branch is returning `None` for this case around here:

https://github.com/vyperlang/vyper/blob/2f1584bf9d2fce15e2173b8befc83619c5deae79/vyper/old_codegen/expr.py#L392-L405


By adding `codehash` to the list, this at least makes it to "reserved keyword error".
I believe `balance` and `codesize` already behave similarly, so I thought it's fine to this way.

Please let me know what's the preferred approach here.

Reply:
This seems OK

Reply:
Yeah seems alright to add, just didn't know if other code was needed to support it.